test/ServiceBus.Tests/SlowConsumingTests/EHSlowConsumingTests.cs

```diff
@@ -0,0 +1,178 @@
+        private const double flowControlThredhold = 0.6;
+        public static readonly EventHubStreamProviderSettings ProviderSettings =
+            new EventHubStreamProviderSettings(StreamProviderName);
+
+        private static readonly Lazy<EventHubSettings> EventHubConfig = new Lazy<EventHubSettings>(() =>
+            new EventHubSettings(
+                TestDefaultConfiguration.EventHubConnectionString,
+                EHConsumerGroup, EHPath));
+
+        private static readonly EventHubCheckpointerSettings CheckpointerSettings =
+            new EventHubCheckpointerSettings(TestDefaultConfiguration.DataConnectionString,
+                EHCheckpointTable, CheckpointNamespace, TimeSpan.FromSeconds(1));
+
+
+        private readonly Fixture fixture;
+
+        public class Fixture : BaseTestClusterFixture
+        {
+            protected override TestCluster CreateTestCluster()
+            {
+                var options = new TestClusterOptions(2);
+                ProviderSettings.SlowConsumingMonitorFlowControlCheckperiod = monitorCheckperiod;
+                ProviderSettings.SlowConsumingMonitorFlowControlThreshold = flowControlThredhold;
+                ProviderSettings.AveragingCachePressureMonitorFlowControlThreshold = null;
+                AdjustClusterConfiguration(options.ClusterConfiguration);
+                return new TestCluster(options);
+            }
+
+            public override void Dispose()
+            {
+                base.Dispose();
+                var dataManager = new AzureTableDataManager<TableEntity>(CheckpointerSettings.TableName, CheckpointerSettings.DataConnectionString);
+                dataManager.InitTableAsync().Wait();
+                dataManager.ClearTableAsync().Wait();
+            }
+
+            private static void AdjustClusterConfiguration(ClusterConfiguration config)
+            {
+                var settings = new Dictionary<string, string>();
+                // get initial settings from configs
+                ProviderSettings.WriteProperties(settings);
+                EventHubConfig.Value.WriteProperties(settings);
+                CheckpointerSettings.WriteProperties(settings);
+
+                // add queue balancer setting
+                settings.Add(PersistentStreamProviderConfig.QUEUE_BALANCER_TYPE, StreamQueueBalancerType.DynamicClusterConfigDeploymentBalancer.ToString());
+
+                // register stream provider
+                config.Globals.RegisterStreamProvider<EHStreamProviderWithCreatedCacheList>(StreamProviderName, settings);
+                config.Globals.RegisterStorageProvider<MemoryStorage>("PubSubStore");
+            }
+        }
+
+        public EHSlowConsumingTests(Fixture fixture)
+        {
+            this.fixture = fixture;
+        }
+
+        [Fact, TestCategory("EventHub"), TestCategory("Streaming"), TestCategory("Functional")]
+        public async Task EHSlowConsuming_ShouldFavorSlowConsumer()
```

Comment:
@jdom I think I need to skip this test if azure is not set up correctly in Jenkins? 

I got this when running in Jenkins:
MESSAGE:
System.AggregateException : One or more errors occurred.\r\n---- System.TypeInitializationException : The type initializer for 'ServiceBus.Tests.SlowConsumingTests.EHSlowConsumingTests' threw an exception.\r\n-------- System.ArgumentNullException : Value cannot be null.\r\nParameter name: dataConnectionString\r\n---- The following constructor parameters did not have matching fixture data: Fixture fixture
+++++++++++++++++++
STACK TRACE:
----- Inner Stack Trace #1 (System.TypeInitializationException) ----- at ServiceBus.Tests.SlowConsumingTests.EHSlowConsumingTests.Fixture.CreateTestCluster() in D:\j\workspace\netfx-functio---224c30ba\test\ServiceBus.Tests\SlowConsumingTests\EHSlowConsumingTests.cs:line 57 at TestExtensions.BaseTestClusterFixture..ctor() in D:\j\workspace\netfx-functio---224c30ba\test\TestExtensions\BaseClusterFixture.cs:line 30 ----- Inner Stack Trace ----- at Orleans.ServiceBus.Providers.EventHubCheckpointerSettings..ctor(String dataConnectionString, String table, String checkpointNamespace, Nullable`1 persistInterval) in D:\j\workspace\netfx-functio---224c30ba\src\OrleansServiceBus\Providers\Streams\EventHub\ICheckpointerSettings.cs:line 65 at ServiceBus.Tests.SlowConsumingTests.EHSlowConsumingTests..cctor() in D:\j\workspace\netfx-functio---224c30ba\test\ServiceBus.Tests\SlowConsumingTests\EHSlowConsumingTests.cs:line 45 ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----

This test pass when I have connection string set up locally